e2e/cypress/integration/system_console/site_configuration/announcement_banner_spec.js

```diff
@@ -0,0 +1,107 @@
+// ***************************************************************
+
+import {hexToRgbArray, rgbArrayToString} from '../../../utils';
+
+describe('Announcement Banner', () => {
+    it('MM-T1128 Announcement Banner - Dismissible banner shows long text truncated', () => {
+        const bannerText =
+            "Here's an announcement! It has a link: http://example.com. It's a really long announcement, because we have a lot to say. Be sure to read it all, click the link, then dismiss the banner, and then you can go on to the next test, which will have a shorter announcement. Thank you for reading and have a nice day!";
+        const bannerBgColor = '#4378da';
+        const bannerBgColorRGBArray = hexToRgbArray(bannerBgColor);
+        const bannerTextColor = '#ffffff';
+        const bannerTextColorRGBArray = hexToRgbArray(bannerTextColor);
+
+        // # Login as System Admin
+        cy.apiAdminLogin();
+
+        // # Go to announcement banner page of system console
+        cy.visit('/admin_console/site_config/announcement_banner');
+
+        // # Enable banner if not already enabled
+        cy.findByTestId('AnnouncementSettings.EnableBanner').within(() => {
+            cy.findByText('true').should('be.visible').click({force: true});
+        });
+
+        // # Enter the long banner text
+        cy.findByTestId('AnnouncementSettings.BannerText').within(() => {
+            cy.get('input').should('be.visible').clear().type(bannerText);
+        });
+
+        // # Change the banner background color
+        cy.findByTestId('AnnouncementSettings.BannerColor').within(() => {
+            cy.get('input').should('be.visible').clear().type(bannerBgColor);
+        });
+
+        // # Change the banner text color
+        cy.findByTestId('AnnouncementSettings.BannerTextColor').within(() => {
+            cy.get('input').should('be.visible').clear().type(bannerTextColor);
+        });
+
+        // # Allow for banner dismissal to true
+        cy.findByTestId('AnnouncementSettings.AllowBannerDismissal').within(
+            () => {
+                cy.findByText('true').
+                    should('be.visible').
+                    click({force: true});
+            },
+        );
+
+        // # Click on the save button
+        cy.get('.admin-console').within(() => {
+            cy.findByText('Save').should('be.visible').click();
+        });
+
+        // * Verify banner text overflows, and its background and color matches to configuration entered before
+        cy.get('.announcement-bar').
+            should('exist').
+            and('have.css', 'overflow', 'hidden').
+            and(
+                'have.css',
+                'background-color',
+                rgbArrayToString(bannerBgColorRGBArray),
+            ).
+            and(
+                'have.css',
+                'color',
+                rgbArrayToString(bannerTextColorRGBArray),
+            );
```

Comment:
i missed one point from spec
> URL is a live link
i think if i just prove that anchor tag with href exists ?

Reply:
added the url test too